main.go

```diff
@@ -49,19 +49,43 @@ func main() {
 		logger.Fatal("creating metrics clientset", zap.Error(err))
 	}
 
-	podMetricsAccessor := clientset.MetricsV1beta1().PodMetricses(config.Namespace)
+	handler := pkg.Handler{
+		Logger:         logger,
+		ExporterConfig: config,
+		KubeConfig:     kubeConfig,
+		ClientSet:      clientset,
+	}
 
-	err = prometheus.Register(pkg.NewKubeMetricsCollector(podMetricsAccessor, logger.Named("collector"), config.Namespace))
+	namespaces, err = handler.GetNamespaces()
 	if err != nil {
-		logger.Fatal("collector registration failed", zap.Error(err))
+		logger.Fatal("failed getting namespaces", zap.Error(err))
 	}
 
+	err = handler.SetupCollector(namespaces)
+	if err != nil {
+		logger.Fatal("could not register collectors", zap.Error(err))
+	}
+
+	go func() {
+		for {
+			logger.Info("checking namespaces...")
+			time.Sleep(time.Second * 20)
+			mu.Lock()
+			namespaces, err = handler.GetNamespaces()
+			if err != nil {
+				logger.Fatal("failed getting namespaces", zap.Error(err))
```

Comment:
if I read this right, this would kill the whole thing if getting the namespaces fails once? Maybe better to just log this error and try again next time.